org.osgi.test.common/src/main/java/org/osgi/test/common/context/CloseableBundleContext.java

```diff
@@ -237,7 +237,11 @@ public ClosableServiceObjects(ServiceObjects<S> so) {
 
 		@Override
 		public void close() {
-			instances.forEach(so::ungetService);
+			instances.forEach((service, useCount) -> {
+				for (int i = 0; i < useCount; i++) {
```

Comment:
We should explicitly unbox `useCount`:

```java
for (int i = useCount; i > 0; i--) {
  so.ungetService(service);
}
```

Reply:
Ok. (The compiler is not smart enough to do that?)

Reply:
Never trust a compiler when you can do the right thing easily yourself :-)